### DISCUSSION TITLE
TS0601 Attribute limits

### DISCUSSION THREAD

**Discussion by jimmyburnworld** [Koenkk/zigbee2mqtt#15497] [Devices]
I've used Z2M for the purpose of controlling some Tuya type radiator thermostats (TRV). My particular model is a Kekotek KTF0177 from Amazon in the UK. When paired it shows as a Tuya 'TS0601_thermostat_3' and exposes all the attributes I require for now.

I'm using a separate Zigbee temperature sensor in the room as I've found measurement at the TRV is inaccurate. I would like to use the 'calibration adjust' method detailed here: [https://github.com/Koenkk/zigbee2mqtt/discussions/5748#discussioncomment-286587](url) which is ideal for my application.

The issue I have is that the 'local_temperature_calibration' attribute is limited to -3 to +3. If I try to set a value from a script <-3 it gives an error. Interestingly in the 'Exposes' window of the Z2M interface, if I set the value to -1, and then type a 0 after, it can be set to -10. The TRV then reports a temperature with a -10 adjustment so I know the device can accept a value of less than -3.

Is there anyway of removing the -3 to +3 limit from the attribute?

**Comment by JuMi2006** [comment]
Opened an issue: 
https://github.com/Koenkk/zigbee2mqtt/issues/16783
